erpnext/accounts/doctype/dunning/dunning.py

```diff
@@ -0,0 +1,121 @@
+from six import string_types
+from frappe.utils import getdate, get_datetime, rounded, flt
+from erpnext.loan_management.doctype.loan_interest_accrual.loan_interest_accrual import days_in_year
+from erpnext.accounts.general_ledger import make_gl_entries, make_reverse_gl_entries
+from erpnext.accounts.doctype.accounting_dimension.accounting_dimension import get_accounting_dimensions
+from erpnext.controllers.accounts_controller import AccountsController
+
+
+class Dunning(AccountsController):
+	def validate(self):
+		self.validate_overdue_days()
+		self.validate_amount()
+
+	def validate_overdue_days(self):
+		self.overdue_days = (getdate(self.posting_date) - getdate(self.due_date)).days or 0
+
+	def validate_amount(self):
+		amounts = calculate_interest_and_amount(
+			self.posting_date, self.outstanding_amount, self.rate_of_interest, self.dunning_fee, self.overdue_days)
+		if self.interest_amount != amounts.get('interest_amount'):
+			self.interest_amount = amounts.get('interest_amount')
+		if self.dunning_amount != amounts.get('dunning_amount'):
+			self.dunning_amount = amounts.get('dunning_amount')
+		if self.grand_total != amounts.get('grand_total'):
+			self.grand_total = amounts.get('grand_total')
+
+	def on_submit(self):
+		self.make_gl_entries()
+
+	def on_cancel(self):
+		if self.dunning_amount:
+			self.ignore_linked_doctypes = ('GL Entry', 'Stock Ledger Entry')
+			make_reverse_gl_entries(voucher_type=self.doctype, voucher_no=self.name)
+
+	def make_gl_entries(self):
+		if not self.dunning_amount:
+			return
+		company_currency = frappe.get_cached_value('Company',  self.company, "default_currency")
+		gl_entries = []
+		invoice_fields = ["project", "cost_center", "debit_to", "against_income_account",
+			"party_account_currency", "conversion_rate", "cost_center"]
+
+		inv = frappe.db.get_value("Sales Invoice", self.sales_invoice, invoice_fields, as_dict=1)
+		accounting_dimensions = get_accounting_dimensions()
+
+		invoice_fields.extend(accounting_dimensions)
+
+		dunning_in_company_currency = flt(self.dunning_amount * inv.conversion_rate)
+
+		gl_entries.append(
+			self.get_gl_dict({
+				"account": inv.debit_to,
+				"party_type": "Customer",
+				"party": self.customer,
+				"due_date": self.due_date,
+				"against": inv.against_income_account,
+				"debit": self.dunning_amount,
+				"debit_in_account_currency": dunning_in_company_currency \
+					if inv.party_account_currency==company_currency else self.dunning_amount,
+				"against_voucher": self.name,
+				"against_voucher_type": "Dunning",
+				"cost_center": inv.cost_center,
+				"project": inv.project
+			}, inv.party_account_currency, item=inv)
+		)
+		gl_entries.append(
+			self.get_gl_dict({
+				"account": inv.against_income_account,
+				"against": self.customer,
+				"credit": self.dunning_amount,
+				"cost_center": inv.cost_center
+			}, item=inv)
+		)
+		make_gl_entries(gl_entries, cancel=(self.docstatus == 2), update_outstanding="Yes", merge_entries=False)
```

Comment:
```suggestion
		make_gl_entries(gl_entries, cancel=(self.docstatus == 2), update_outstanding="No", merge_entries=False)
```

As we are not maintaining outstanding for Dunning, it should not be updated via GL Entry.